src/containers/Stage.js

```diff
@@ -1,71 +1,75 @@
 import React, { Component } from 'react';
 import { bindActionCreators } from 'redux';
 import { connect } from 'react-redux';
+import { Link } from 'react-router';
 
 import { actionCreators as stageActions } from '../ducks/modules/stage';
+import { getStages } from '../selectors/session';
 
 import { NameGenerator } from './Interfaces';
 
 /**
-  * Render a protocol interface based on protocol info and index
+  * Render a protocol interface based on protocol info and id
   */
 class Stage extends Component {
   constructor(props) {
     super(props);
-
-    this.handleNext = this.handleNext.bind(this);
-    this.handlePrevious = this.handlePrevious.bind(this);
-  }
-
-  handleNext() {
-    this.props.next();
-  }
-
-  handlePrevious() {
-    this.props.previous();
   }
 
   render() {
     // TODO: Load dynamically based on protocol using some kind of service?
     const CurrentInterface = NameGenerator;
 
-    const {
-      handleNext,
-      handlePrevious,
-      props: {
-        stage
-      }
-    } = this;
-
     return (
       <div className="stage">
         <div className="stage__control">
-          <button className="stage__control-button stage__control-button--back" onClick={handlePrevious}>Back</button>
+          <Link
+            to={'/protocol/' + this.props.prevLink}
+            className="stage__control-button stage__control-button--back"
+            onClick={() => this.props.onStageClick(this.props.stages, this.props.prevLink)}
+          >
+            Back
+          </Link>
         </div>
         <div className="stage__interface">
-          <CurrentInterface config={ stage }/>
+          <CurrentInterface config={ this.props.stage }/>
```

Comment:
Maybe this should be a new PR then? It affects NameGenerator which is not yet altered in this branch.